apps/openmw/mwsound/soundmanagerimp.cpp

```diff
@@ -858,27 +857,36 @@ namespace MWSound
     }
 
 
-    void SoundManager::pauseSounds(int types)
+    void SoundManager::pauseSounds(const std::string& blockerId, int types)
     {
         if(mOutput->isInitialized())
         {
+            auto found = mPausedSoundTypes.find(blockerId);
+            if (found != mPausedSoundTypes.end() && found->second != types)
+                resumeSounds(blockerId);
+
             types = types & Type::Mask;
             mOutput->pauseSounds(types);
-            mPausedSoundTypes |= types;
+            mPausedSoundTypes[blockerId] = types;
```

Comment:
Won't this unpause all sounds currently paused by `blockerId`, then repause what's being set? The given types should combine with what's already there for the given blocker, or at the very least, shouldn't unpause sounds that are going to get paused again.

Reply:
> Won't this unpause all sounds currently paused by blockerId, then repause what's being set?

It is for a quite specific case when user set a different mask for already used blocker. Yes, they the only readable approach I was able to get working here is to "remove an old blocker and set a new one".